presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraMetadata.java

```diff
@@ -86,17 +85,18 @@ public CassandraMetadata(CassandraConnectorId connectorId,
     @Override
     public List<String> listSchemaNames(ConnectorSession session)
     {
-        return schemaProvider.getAllSchemas();
+        return cassandraSession.getCaseSensitiveSchemaNames().stream()
```

Comment:
I don't want to fail as long as possible. For example if you have 2 keyspaces that are spelled the same way in lower case you still can query other keyspaces, that doesn't clash in name. The similar approach has been chosen for the table names. The only place where we cannot behave the same way - are the column names. Because presto internally uses a map for column handles.